src/core/Akka.Remote/Serialization/DaemonMsgCreateSerializer.cs

```diff
@@ -114,6 +114,17 @@ public override object FromBinary(byte[] bytes, Type type)
         {
             var proto = DaemonMsgCreateData.ParseFrom(bytes);
             var clazz = Type.GetType(proto.Props.Clazz);
+
+            if (clazz == null)
+            {
+                var msg =
+                    string.Format(
+                        "Could not find '{0}' in loaded assemblies. If this is a remote/cluster deploy scenario, " +
+                        "ensure the remote system has referenced the assembly containing this type.", proto.Props.Clazz);
+                    
+                throw new ActorInitializationException(msg);
```

Comment:
I think ```TypeLoadException``` is perfectly fine exception as is. IMO, most .NET developers would know exactly what was going on if they ran across that exception.

But if you wanted it to be more user friendly, maybe create a custom exception like RemoteTypeLoadException and then do something like

```csharp
Type clazz;
try
{
    clazz = Type.GetType(proto.Props.Clazz, true);
}
catch(TypeLoadException tle)
{
    var msg = string.Format(
        "Could not find type '{0}' on the remote system. Ensure that the remote system has an assembly that contains {0} in its assembly search path", proto.Props.Clazz);

    throw new RemoteTypeLoadException(msg, tle);
}
```

or have our custom RemoteTypeLoadException already have msg's verbiage and just plug in the {0} value as a property on that exception.

```csharp
throw new RemoteTypeLoadException(tle) { Type = proto.Props.Clazz };
```